test/prob/utility.hpp

```diff
@@ -54,13 +54,69 @@ std::ostream& operator<<(std::ostream& os, const vector<var>& param) {
 }  // namespace std
 
 // ------------------------------------------------------------
+
+template <typename T>
+T get_param(const vector<double>& params, const size_t n) {
+  T param = 0;
+  if (n < params.size())
+    param = params[n];
+  return param;
+}
+
+template <>
+empty get_param<empty>(const vector<double>& /*params*/, const size_t /*n*/) {
+  return empty();
+}
+
+template <>
+fvar<double> get_param<fvar<double>>(const vector<double>& params,
+                                     const size_t n) {
+  fvar<double> param = 0;
+  if (n < params.size()) {
+    param = params[n];
+    param.d_ = 1.0;
+  }
+  return param;
+}
+template <>
+fvar<var> get_param<fvar<var>>(const vector<double>& params, const size_t n) {
+  fvar<var> param = 0;
+  if (n < params.size()) {
+    param = params[n];
+    param.d_ = 1.0;
+  }
+  return param;
+}
+template <>
+fvar<fvar<double>> get_param<fvar<fvar<double>>>(const vector<double>& params,
+                                                 const size_t n) {
+  fvar<fvar<double>> param = 0;
+  if (n < params.size()) {
+    param = params[n];
+    param.d_ = 1.0;
+  }
+  return param;
+}
+template <>
+fvar<fvar<var>> get_param<fvar<fvar<var>>>(const vector<double>& params,
+                                           const size_t n) {
+  fvar<fvar<var>> param = 0;
+  if (n < params.size()) {
+    param = params[n];
+    param.d_ = 1.0;
+  }
+  return param;
+}
+
+// ------------------------------------------------------------
+
 // default template handles Eigen::Matrix
 template <typename T>
 T get_params(const vector<vector<double>>& parameters, const size_t p) {
   T param(parameters.size());
   for (size_t n = 0; n < parameters.size(); n++)
     if (p < parameters[0].size())
-      param(n) = parameters[n][p];
+      param(n) = get_param<stan::scalar_type_t<T>>(parameters[n], p);
```

Comment:
For higher order autodiff types, we need to initialize each of the params with something more than just casting up a double.

For vars, we assign the derivative term to 1.0, for instance.

If we don't do this code that depends on `get_params` and `get_param` getting the same params fails.